fvtest/compilerunittest/abstractinterpreter/AbsInterpreterTest.cpp

```diff
@@ -0,0 +1,155 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include <gtest/gtest.h>
+#include <exception>
+#include "../CodeGenTest.hpp"
+#include "AbsInterpreterTest.hpp"
+#include "optimizer/abstractinterpreter/AbsValue.hpp"
+#include "optimizer/abstractinterpreter/AbsOpArray.hpp"
+#include "optimizer/abstractinterpreter/AbsOpStack.hpp"
+
+class AbsVPValueTest : public TRTest::AbsInterpreterTest {};
+
+TEST_F(AbsVPValueTest, testParameter) {
+   TR::VPIntConst constraint(1);
+   TR::AbsVPValue value1(vp(), &constraint, TR::Int32);
+   ASSERT_FALSE(value1.isParameter());
+   value1.setParamPosition(1);
+   ASSERT_TRUE(value1.isParameter());
+}
+
+TEST_F(AbsVPValueTest, testSetToTop) {
+   TR::VPIntConst constraint(1);
+   TR::AbsVPValue value1(vp(), &constraint, TR::Int32);
+   ASSERT_FALSE(value1.isTop());
+   value1.setToTop();
+   ASSERT_TRUE(value1.isTop());
+}
+
+TEST_F(AbsVPValueTest, testDataType) {
+   TR::VPIntConst constraint(1);
+   TR::AbsVPValue value1(vp(), &constraint, TR::Int32);
+   ASSERT_FALSE(value1.isTop());
+   value1.setToTop();
+   ASSERT_TRUE(value1.isTop());
+}
+
+TEST_F(AbsVPValueTest, testSetParam) {
+   TR::VPIntConst constraint(1);
+   TR::AbsVPValue value1(vp(), &constraint, TR::Int32);
+   bool dataTypeEq = false;
+   if (value1.getDataType() == TR::Int32)
+      dataTypeEq = true;
+   ASSERT_TRUE(dataTypeEq);
```

Comment:
This is because the == operator of TR::DataType is not declared as const, but ASSERT_TRUE requires comparison operator be const, so there is compilation error if using ASSERT_EQ directly.